### DISCUSSION TITLE
When does minio delete objects on disk?

### DISCUSSION THREAD

**Discussion by MiniCodeMonkey** [minio/minio#17225] [Q&A] [answered]
Hi there!

I'm running a simple single-instance minio server. When I delete an object in a bucket, it is kept on the disk and seems to stay there for weeks.

Is there a manual "garbage collection" routine that I can invoke from the console to remove these old objects on disk?

Mathias

---

I'm running minio as a Docker image with `RELEASE.2023-05-04T21-44-30Z`

**Comment by klauspost** [comment]
Objects should be moved to `.minio.sys/tmp/.trash` immediately, unless replication is configured. This should then be deleted at a reasonable pace.

How are you observing the files staying?

**Comment by MiniCodeMonkey** [comment]
Hej Klaus!

Thanks for providing some additional context around the deletion process. The files are staying in place on disk, but are not showing up with `mc ls` or in the object browser. They appear to be soft deleted.

I just checked, and I had an old, disabled replication rule in place. I just tried deleting it altogether and will wait and see if that resolved it.

**Reply by harshavardhana** [reply, marked as answer]
> Hej Klaus!
> 
> Thanks for providing some additional context around the deletion process. The files are staying in place on disk, but are not showing up with `mc ls` or in the object browser. They appear to be soft deleted.
> 
> I just checked, and I had an old, disabled replication rule in place. I just tried deleting it altogether and will wait and see if that resolved it.

If you want to remove permanently you need to use --versions via `mc rm`

**Reply by MiniCodeMonkey** [reply]
> > Hej Klaus!
> > Thanks for providing some additional context around the deletion process. The files are staying in place on disk, but are not showing up with `mc ls` or in the object browser. They appear to be soft deleted.
> > I just checked, and I had an old, disabled replication rule in place. I just tried deleting it altogether and will wait and see if that resolved it.
> 
> If you want to remove permanently you need to use --versions via `mc rm`

Bingo! That was probably it. Doh!

Thanks!
